File: Tutorials/ComboBoxExample/ComboBoxJVM/src/main/java/ComboBoxExample/ItemsFactory.java
package ComboBoxExample;

import java.awt.Color;
import java.awt.Font;
import java.awt.image.BufferedImage;

import com.spvessel.spacevil.ButtonCore;
import com.spvessel.spacevil.Ellipse;
import com.spvessel.spacevil.ImageItem;
import com.spvessel.spacevil.MenuItem;
import com.spvessel.spacevil.PopUpMessage;
import com.spvessel.spacevil.Decorations.Indents;
import com.spvessel.spacevil.Decorations.ItemState;
import com.spvessel.spacevil.Flags.InputEventType;
import com.spvessel.spacevil.Flags.ItemAlignment;
import com.spvessel.spacevil.Flags.ItemStateType;
import com.spvessel.spacevil.Flags.SizePolicy;

final class ItemsFactory {
    private ItemsFactory() {
    }

    static MenuItem getMenuItem(String name, BufferedImage bitmap) {
        MenuItem menuItem = new MenuItem(name);
        menuItem.setStyle(StyleFactory.getMenuItemStyle());
        menuItem.setTextMargin(new Indents(25, 0, 0, 0));

        // Optionally: set an event on click
        menuItem.eventMouseClick.add((sender, args) -> {
            PopUpMessage popUpInfo = new PopUpMessage("You choosed a function:\n" + menuItem.getText());
            popUpInfo.setStyle(StyleFactory.getBluePopUpStyle());
            popUpInfo.setTimeOut(2000);
            popUpInfo.show(menuItem.getHandler());
        });

        // Optionally: add an image into MenuItem
        ImageItem img = new ImageItem(bitmap, false);
        img.setSizePolicy(SizePolicy.FIXED, SizePolicy.FIXED);
        img.setSize(20, 20);
        img.setAlignment(ItemAlignment.LEFT, ItemAlignment.VCENTER);
        img.keepAspectRatio(true);
        menuItem.addItem(img);

        // Optionally: add a button into MenuItem
        ButtonCore infoBtn = new ButtonCore("?");
        infoBtn.setBackground(40, 40, 40);
        infoBtn.setWidth(20);
        infoBtn.setSizePolicy(SizePolicy.FIXED, SizePolicy.EXPAND);
        infoBtn.setFontStyle(Font.BOLD);
        infoBtn.setForeground(210, 210, 210);
        infoBtn.setAlignment(ItemAlignment.VCENTER, ItemAlignment.RIGHT);
        infoBtn.setMargin(0, 0, 10, 0);
        infoBtn.setBorderRadius(3);
        infoBtn.addItemState(ItemStateType.HOVERED, new ItemState(new Color(10, 140, 210)));
        infoBtn.setPassEvents(false, InputEventType.MOUSE_PRESS, InputEventType.MOUSE_RELEASE,
                InputEventType.MOUSE_DOUBLE_CLICK);
        infoBtn.isFocusable = false; // prevent focus this button
        infoBtn.eventMouseClick.add((sender, args) -> {
            PopUpMessage popUpInfo = new PopUpMessage("This is decorated MenuItem:\n" + menuItem.getText());
            popUpInfo.setStyle(StyleFactory.getDarkPopUpStyle());
            popUpInfo.setTimeOut(2000);
            popUpInfo.show(infoBtn.getHandler());
        });
        menuItem.addItem(infoBtn);

        return menuItem;
    }

    static Ellipse getDot() {
        Ellipse ellipse = new Ellipse(12);
        ellipse.setSize(8, 8);
        ellipse.setAlignment(ItemAlignment.VCENTER, ItemAlignment.LEFT);
        ellipse.setMargin(10, 0, 0, 0);
        return ellipse;
    }
}

File: Tutorials/ComboBoxExample/ComboBoxNet/ItemsFactory.cs
using System;
using System.Drawing;
using SpaceVIL;
using SpaceVIL.Core;
using SpaceVIL.Decorations;

namespace ComboBoxExample
{
    internal static class ItemsFactory
    {
        internal static MenuItem GetMenuItem(String name, Bitmap bitmap)
        {
            MenuItem menuItem = new MenuItem(name);
            menuItem.SetStyle(StyleFactory.GetMenuItemStyle());
            menuItem.SetTextMargin(new Indents(25, 0, 0, 0));


            // Optionally: set an event on click
            menuItem.EventMouseClick += (sender, args) =>
            {
                PopUpMessage popUpInfo = new PopUpMessage("You choosed a function:\n" + menuItem.GetText());
                popUpInfo.SetStyle(StyleFactory.GetBluePopUpStyle());
                popUpInfo.SetTimeOut(2000);
                popUpInfo.Show(menuItem.GetHandler());
            };

            // Optionally: add an image into MenuItem
            ImageItem img = new ImageItem(bitmap, false);
            img.SetSizePolicy(SizePolicy.Fixed, SizePolicy.Fixed);
            img.SetSize(20, 20);
            img.SetAlignment(ItemAlignment.Left, ItemAlignment.VCenter);
            img.KeepAspectRatio(true);
            menuItem.AddItem(img);


            // Optionally: add a button into MenuItem
            ButtonCore infoBtn = new ButtonCore("?");
            infoBtn.SetBackground(40, 40, 40);
            infoBtn.SetWidth(20);
            infoBtn.SetSizePolicy(SizePolicy.Fixed, SizePolicy.Expand);
            infoBtn.SetFontStyle(FontStyle.Bold);
            infoBtn.SetForeground(210, 210, 210);
            infoBtn.SetAlignment(ItemAlignment.VCenter, ItemAlignment.Right);
            infoBtn.SetMargin(0, 0, 10, 0);
            infoBtn.SetBorderRadius(3);
            infoBtn.AddItemState(ItemStateType.Hovered, new ItemState(Color.FromArgb(0, 140, 210)));
            infoBtn.SetPassEvents(false, InputEventType.MousePress, InputEventType.MouseRelease, InputEventType.MouseDoubleClick);
            infoBtn.IsFocusable = false; // prevent focus this button
            infoBtn.EventMouseClick += (sender, args) =>
            {
                PopUpMessage popUpInfo = new PopUpMessage("This is decorated MenuItem:\n" + menuItem.GetText());
                popUpInfo.SetStyle(StyleFactory.GetDarkPopUpStyle());
                popUpInfo.SetTimeOut(2000);
                popUpInfo.Show(infoBtn.GetHandler());
            };
            menuItem.AddItem(infoBtn);

            return menuItem;
        }

        internal static Ellipse GetDot()
        {
            Ellipse ellipse = new Ellipse(12);
            ellipse.SetSize(8, 8);
            ellipse.SetAlignment(ItemAlignment.VCenter, ItemAlignment.Left);
            ellipse.SetMargin(10, 0, 0, 0);
            return ellipse;
        }
    }
}